### PR TITLE
Fix places where std::plus<T> used without std::plus<void>

### PR DESCRIPTION
**Pull request by SergeyKopienko**
In this PR we fix some places where ```std::plus<T>``` used ```without std::plus<void>```

### CODE REVIEW

File: include/oneapi/dpl/pstl/numeric_impl.h [hunk 1]
```diff
@@ -194,7 +194,8 @@ __brick_transform_scan(_RandomAccessIterator __first, _RandomAccessIterator __la
 template <typename _Tp, typename _BinaryOperation>
 using is_arithmetic_udop =
     ::std::integral_constant<bool, ::std::is_arithmetic<_Tp>::value &&
-                                       !::std::is_same<_BinaryOperation, ::std::plus<_Tp>>::value>;
+                                       !::std::is_same<_BinaryOperation, ::std::plus<_Tp>>::value &&
```
Comment (MikeDvorskiy) [inline include/oneapi/dpl/pstl/numeric_impl.h:197]: What's about `__dpl_sycl::__plus` ?

Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/numeric_impl.h:197]: This predicate implemented in our code as
```C++
#if _ONEDPL_SYCL2020_FUNCTIONAL_OBJECTS_PRESENT
template <typename _T = void>
using __plus = sycl::plus<_T>;
#else  // _ONEDPL_SYCL2020_FUNCTIONAL_OBJECTS_PRESENT
template <typename _T>
using __plus = sycl::ONEAPI::plus<_T>;
#endif // _ONEDPL_SYCL2020_FUNCTIONAL_OBJECTS_PRESENT
```

where ```sycl::plus<_T>``` implemented as
```C++
template <typename T = void> using plus = std::plus<T>;
```

Comment (adamfidel) [inline include/oneapi/dpl/pstl/numeric_impl.h:197]: I notice in some places you are using `is_same` and in others `is_same_v`. Could they all be `is_same_v`?

Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/numeric_impl.h]: @MikeDvorskiy Fixed at https://github.com/oneapi-src/oneDPL/pull/1125/commits/3f3e0d908c7af03f870b5f7c5724558dc4b4fc19 : is it that you mean?

Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/numeric_impl.h]: @adamfidel My proposal to ignore this moment at current PR.
I have been prepared separate PR: https://github.com/oneapi-src/oneDPL/pull/1107
May be we apply this PR later.

File: include/oneapi/dpl/pstl/numeric_impl.h [hunk 2]
```diff
@@ -27,6 +27,10 @@
 
 #include "parallel_backend.h"
 
+#if _ONEDPL_BACKEND_SYCL
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/numeric_impl.h]: Agree, fixed.

File: include/oneapi/dpl/pstl/unseq_backend_simd.h [hunk 3]
```diff
@@ -20,6 +20,10 @@
 
 #include "utils.h"
 
+#if _ONEDPL_BACKEND_SYCL
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/unseq_backend_simd.h]: Fixed.

File: include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h [hunk 4]
```diff
@@ -76,7 +76,10 @@ using __has_known_identity = std::false_type;
 template <typename _BinaryOp, typename _Tp>
 struct __known_identity_for_plus
 {
-    static_assert(std::is_same_v<typename std::decay<_BinaryOp>::type, std::plus<_Tp>>);
+    static_assert(std::is_same_v<typename std::decay<_BinaryOp>::type, std::plus<_Tp>> ||
+                  std::is_same_v<typename std::decay<_BinaryOp>::type, std::plus<void>> ||
```
Comment (MikeDvorskiy) [inline include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h:80]: std -> ::std

File: include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h [hunk 5]
```diff
@@ -76,7 +76,10 @@ using __has_known_identity = std::false_type;
 template <typename _BinaryOp, typename _Tp>
 struct __known_identity_for_plus
 {
-    static_assert(std::is_same_v<typename std::decay<_BinaryOp>::type, std::plus<_Tp>>);
+    static_assert(std::is_same_v<typename std::decay<_BinaryOp>::type, std::plus<_Tp>> ||
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h]: Fixed

File: include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h (hunk 4 above)
Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h]: Fixed

**Review by MikeDvorskiy — APPROVED** [review]
LGTM

**Pull request merged**